lib/collection/src/shards/transfer/mod.rs

```diff
@@ -61,7 +61,7 @@ pub struct ShardTransferKey {
 
 impl ShardTransferKey {
     pub fn check(&self, transfer: &ShardTransfer) -> bool {
-        self.shard_id == transfer.shard_id && self.from == transfer.from && self.to == transfer.to
+        transfer.key() == *self
     }
```

Comment:
Can't "suggest" on unmodified lines, but as `ShardTransferKey` is `Copy` now:

```rust
pub fn check(self, transfer: &ShardTransfer) -> bool {
  self == transfer.key()
}
```

Reply:
Good catch removing the &. Thanks!